ludwig/features/feature_utils.py

```diff
@@ -52,3 +56,74 @@ def sanitize(name):
 def compute_feature_hash(feature: dict) -> str:
     preproc_hash = hash_dict(feature.get(PREPROCESSING, {}))
     return sanitize(feature[NAME]) + "_" + preproc_hash.decode("ascii")
+
+
+def get_input_size_with_dependencies(
+    combiner_output_size: int, dependencies: List[str], other_output_features  # Dict[str, "OutputFeature"]
+):
+    """Returns the input size for the first layer of this output feature's FC stack, accounting for dependencies on
+    other output features.
+
+    In the forward pass, the hidden states of any dependent output features get concatenated with the combiner's output.
+    If this output feature depends on other output features, then the input size for this feature's FCStack is the sum
+    of the output sizes of other output features + the combiner's output size.
+    """
+    input_size_with_dependencies = combiner_output_size
+    for feature_name in dependencies:
+        if other_output_features[feature_name].num_fc_layers:
+            input_size_with_dependencies += other_output_features[feature_name].fc_stack.output_shape[-1]
+        else:
+            # 0-layer FCStack. Use the output feature's input size.
+            input_size_with_dependencies += other_output_features[feature_name].input_size
+    return input_size_with_dependencies
+
+
+class LudwigFeatureDict(torch.nn.Module):
+    """Torch ModuleDict wrapper that permits keys with any name.
+
+    Torch's ModuleDict implementation doesn't allow certain keys to be used if they conflict with existing class
+    attributes, e.g.
+
+    > torch.nn.ModuleDict({'type': torch.nn.Module()})  # Raises KeyError.
+
+    This class is a simple wrapper around torch's ModuleDict that mitigates possible conflicts by using a key-suffixing
+    protocol.
+    """
+
+    def __init__(self):
```

Comment:
It seems alright to have less flexibility in the constructor here. The goal isn't to replicate all of the functionality of a ModuleDict, but with a minimal wrapping to prevent name conflicts. In our usage, we are always initializing a LudwigFeatureDict from empty.